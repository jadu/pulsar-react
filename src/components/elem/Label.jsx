import React from 'react';

export default class Label extends React.Component {
  render() {
    return (
      <>
      {this.props.label 
        ? (
          <label className="control__label">
            {this.props.label}
          </label>
        )
        : (
          <span className="control__label"></span>
        )
      }
      </>
    );
  }
}