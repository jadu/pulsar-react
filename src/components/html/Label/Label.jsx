import React from 'react';
import classnames from 'classnames';

export default class Button extends React.Component {

  static defaultProps = {
    className: 'label'
  };

  render() {
    let {
      children,
      className,
      danger,
      info,
      inverse,
      large,
      primary,
      success,
      warning,
      white,
      ...props
    } = this.props;

    let variantClasses = classnames(className, {
      'label--danger': danger,
      'label--info': info,
      'label--inverse': inverse,
      'label--large': large,
      'label--primary': primary,
      'label--success': success,
      'label--warning': warning,
      'label--white': white
    });

    return (
      <React.Fragment>
        <span
          className={variantClasses}
          {...props}
        >
          {children}
        </span>
      </React.Fragment>
    );
  }
}
