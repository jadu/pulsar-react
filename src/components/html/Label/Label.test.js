'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import Label from './Label';

it('renders the base label', () => {
  const tree = render(
    <Label>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the primary variation', () => {
  const tree = render(
    <Label primary>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the info variation', () => {
  const tree = render(
    <Label info>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the success variation', () => {
  const tree = render(
    <Label success>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the warning variation', () => {
  const tree = render(
    <Label warning>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the danger variation', () => {
  const tree = render(
    <Label danger>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the inverse variation', () => {
  const tree = render(
    <Label inverse>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the white variation', () => {
  const tree = render(
    <Label white>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the large variation', () => {
  const tree = render(
    <Label large>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});

it('renders multiple variations', () => {
  const tree = render(
    <Label primary large>foo</Label>
  );
  expect(tree).toMatchSnapshot();
});
