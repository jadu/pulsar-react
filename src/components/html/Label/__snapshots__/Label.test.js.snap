// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiple variations 1`] = `
<span
  class="label label--large label--primary"
>
  foo
</span>
`;

exports[`renders the base label 1`] = `
<span
  class="label"
>
  foo
</span>
`;

exports[`renders the danger variation 1`] = `
<span
  class="label label--danger"
>
  foo
</span>
`;

exports[`renders the info variation 1`] = `
<span
  class="label label--info"
>
  foo
</span>
`;

exports[`renders the inverse variation 1`] = `
<span
  class="label label--inverse"
>
  foo
</span>
`;

exports[`renders the large variation 1`] = `
<span
  class="label label--large"
>
  foo
</span>
`;

exports[`renders the primary variation 1`] = `
<span
  class="label label--primary"
>
  foo
</span>
`;

exports[`renders the success variation 1`] = `
<span
  class="label label--success"
>
  foo
</span>
`;

exports[`renders the warning variation 1`] = `
<span
  class="label label--warning"
>
  foo
</span>
`;

exports[`renders the white variation 1`] = `
<span
  class="label label--white"
>
  foo
</span>
`;
